test/nix/resolv_conf.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, dnsServer }:
+let
+  host_resolv_conf = builtins.path {
+    name = "host-resolv-conf";
+    path = "/etc/resolv.conf";
+  };
+in stdenv.mkDerivation {
+  name = "resolv-conf";
+  buildCommand = ''
+    RESOLV_CONF_FILE="$out/resolv.conf"
+    mkdir -p $out
+
+    check_loopback_dns() {
+      if [ ! -f "${host_resolv_conf}" ]; then
+        return 1
+      fi
+
+      if grep -qE "nameserver\s+127\.0\.0\." "${host_resolv_conf}"; then
+        return 1
+      else
+        return 0
+      fi
+    }
+
+    if [ -n "${dnsServer}" ] && [ "${dnsServer}" != "none" ]; then
+      echo "nameserver ${dnsServer}" > "$RESOLV_CONF_FILE"
+    elif check_loopback_dns; then
```

Comment:
I think renaming `check_loopback_dns` to `is_host_resolve_conf_valid` would make the condition much more clear.